<launch>  
<node name="laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" output="screen">
	<remap from="image" to="/stereo/depth/image_raw"/>
	<remap from="camera_info" to="/stereo/depth/camera_info"/>
	<remap from="frame_id" to="camera"/>
	<param name="output_frame_id" value="camera_depth_frame"/>
	<param name="scan_height" value="1"/>
	<param name="range_min" value="15"/>
	<param name="range_min" value="0.45"/>
</node>


<node pkg="map_server" type="map_saver"
   name="map_saver"
   args="-f /home/turgay/Bag/map1" output="screen">
</node>

<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
	<remap from="scan" to="scan"/>
	<param name="odom_frame" value="odom"/>
	<param name="map_update_interval" value="5.0"/>
	<param name="maxUrange" value="10.0"/> <!-- The maximum usable range of the laser. A beam is cropped to this value. -->
	<param name="maxRange" value="10.0"/> <!-- The maximum range of the sensor. If regions with no obstacles within the range of the sensor should appear as free space in the map, set maxUrange < maximum range of the real sensor <= maxRange.--> 
	<param name="sigma" value="0.05"/>
	<param name="kernelSize" value="1.0"/>
	<param name="lstep" value="0.05"/>
	<param name="astep" value="0.05"/>
	<param name="iterations" value="5"/>
	<param name="lsigma" value="0.075"/>
	<param name="ogain" value="3.0"/>
	<param name="lskip" value="0"/>
	<param name="srr" value="0.01"/>
	<param name="srt" value="0.02"/>
	<param name="str" value="0.01"/>
	<param name="stt" value="0.02"/>
	<param name="linearUpdate" value="0.3"/>     <!-- Process a scan each time the robot translates this far -->
	<param name="angularUpdate" value="0.1"/>   <!--Process a scan each time the robot rotates this far; 0.1 represents 5.7 degress -->
	<param name="temporalUpdate" value="-1.0"/>
	<param name="resampleThreshold" value="0.5"/>
	<param name="particles" value="180"/>
	<param name="xmin" value="-50.0"/>
	<param name="ymin" value="-50.0"/>
	<param name="xmax" value="50.0"/>
	<param name="ymax" value="50.0"/>
	<param name="delta" value="0.05"/>
	<param name="llsamplerange" value="0.01"/>
	<param name="llsamplestep" value="0.01"/>
	<param name="lasamplerange" value="0.005"/>
	<param name="lasamplestep" value="0.005"/>
</node>

</launch>
